ethernet_init/ethernet_init.c

```diff
@@ -61,7 +67,18 @@
 #define CONFIG_ETHERNET_SPI_NUMBER 0
 #endif
 
+/* This enum definition must be aligned with the ETHERNET_SPI_USE_ID* definitions in Kconfig.projbuild */
+typedef enum {
+    SPI_DEV_TYPE_DM9051,
+    SPI_DEV_TYPE_KSZ8851SNL,
+    SPI_DEV_TYPE_W5500,
+    SPI_DEV_TYPE_CH390,
+    SPI_DEV_TYPE_ENC28J60,
+    SPI_DEV_TYPE_LAN865X,
+} spi_eth_dev_type_t;
+
 typedef struct {
+    spi_eth_dev_type_t dev;
```

Comment:
Isn't this a bit of a breaking change? Do we require users now to set this field?